include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -673,13 +681,138 @@ template <typename Lyt, typename TT>
     assert(lyt.num_pos() > 0 && "skeleton needs output cells");
 
     assert(!spec.empty());
-    // all elements in tts must have the same number of variables
+    // all elements in spec must have the same number of variables
     assert(std::adjacent_find(spec.cbegin(), spec.cend(), [](const auto& a, const auto& b)
                               { return a.num_vars() != b.num_vars(); }) == spec.cend());
 
     detail::is_operational_impl<Lyt, TT> p{lyt, spec, params};
 
-    return p.determine_operational_input_patterns();
+    std::set<uint64_t> input_patterns{};
+
+    // all possible input patterns
+    for (auto i = 0u; i < spec.front().num_bits(); ++i)
+    {
+        input_patterns.insert(i);
+    }
+
+    const auto non_op_patterns_and_non_op_reason =
+        p.determine_non_operational_input_patterns_and_non_operationality_reason();
+
+    for (const auto& [input_pattern, _] : non_op_patterns_and_non_op_reason)
+    {
+        input_patterns.erase(input_pattern);
+    }
+
+    return input_patterns;
+}
+
+/**
+ * This function determines all input combinations for which kinks induce the SiDB layout to become non-operational.
+ * This means that the layout is operational if kinks would be accepted.
+ *
+ * @note "kink induced non-operational" refers to the operational status being exclusively caused to be
+ * `NON_OPERATIONAL` by kinks.
```

Comment:
```suggestion
 * @note "Kink induced non-operational" refers to the non-operational status being exclusively caused by kinks with an otherwise correct logic match.
```